js/render/draw_circle.js

```diff
@@ -4,27 +4,19 @@ var browser = require('../util/browser.js');
 
 module.exports = drawCircles;
 
-function drawCircles(painter, layer, posMatrix, tile) {
-    // short-circuit if tile is empty
-    if (!tile.buffers) return;
-
-    posMatrix = painter.translateMatrix(posMatrix, tile, layer.paint['circle-translate'], layer.paint['circle-translate-anchor']);
+function drawCircles(painter, source, layer, coords) {
+    var gl = painter.gl;
```

Comment:
Native [aborts early if it's not the translucent pass.](https://github.com/mapbox/mapbox-gl-native/blob/45c7b472adb78600ab9f9c8f82082d6b0af21ba2/src/mbgl/renderer/painter_circle.cpp#L18-L18) Do we need that here?